### PR TITLE
feat(firestore-multimodal-genai): add custom hook params

### PR DESCRIPTION
**Pull request by cabljac**
This PR adds a custom hook feature to the gemini tasks extension

### CODE REVIEW

**Comment by pr-Mais** [comment]
can we add docs for this feature

File: firestore-multimodal-genai/PREINSTALL.md [hunk 1]
```diff
@@ -67,6 +67,12 @@ This extension will update the state of a document that is being processed withi
 
 Changing the state field of a completed document's status from `COMPLETED` to anything else will retrigger the extension for that document.
 
+### Custom RAG Hook
+
+If you specify a a RAG (Retrieval Augmentation Generation) hook URL during installation, the extension will call this endpoint to obtain additional data for the prompt. You may specify an API key for your hook, which will be passed in the `x-api-key` header. You should also specify input and output fields for the RAG hook, as comma separated lists.
```
Comment (cabljac) [inline firestore-multimodal-genai/PREINSTALL.md:72]: @pr-Mais  I added docs here, do we need more?

Comment (pr-Mais) [reply firestore-multimodal-genai/PREINSTALL.md:72]: Maybe an example?

File: firestore-multimodal-genai/PREINSTALL.md [hunk 2]
```diff
@@ -67,6 +67,20 @@ This extension will update the state of a document that is being processed withi
...
+
+The input fields will be extracted from the document and passed in the body of the request to the RAG endpoint, whereas the response body will be filtered for the output fields, and these will be made available as handlebars variables when the extension creates the prompt for Gemini.
+
+#### Example
+
+Suppose we configure the extension with `RAG_HOOK_URL=https://movieAPI.com/search`, `RAG_HOOK_INPUT_FIELDS=query` and `RAG_HOOK_OUTPUT_FIELDS=searchResult`, and our prompt is `PROMPT=Tell about the following actors in this movie {{searchResult}}`.
+
+The extension will call the RAG hook, extract the string `searchResult` from the JSON body of the respons, and substitute it into the prompt it later calls Gemini with.
```
Comment (cabljac) [inline firestore-multimodal-genai/PREINSTALL.md:80]:
```suggestion
The extension will call the RAG hook, extract the string `searchResult` from the JSON body of the response, and substitute it into the prompt it later calls Gemini with.
```

**Review by pr-Mais — APPROVED** [review]

**Pull request merged**